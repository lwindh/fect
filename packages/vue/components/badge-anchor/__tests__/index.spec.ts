import { mount } from '@vue/test-utils'
import BadgeAnchor from '..'
import Badge from '../../badge/index'

const Wrapper = {
  components:{
    [BadgeAnchor.name]:BadgeAnchor,
    [Badge.name]:Badge,
  },
  data(){
    return {
      place:'topRight',
      dot:false,
    }
  },
  template:`<div class="container">
   <fe-badgeAnchor :placement="place">
    <p>Test Message</p>
    <fe-badge :dot="dot"/>
   </fe-badgeAnchor>
  </div>`,
}


describe('BadgeAnchor', () =>{
  it('should be render as a context ',async()=>{
    const wrapper = mount(Wrapper)
    const el = wrapper.find('sup')
    expect(el.attributes('style')).toBe(
      'position: absolute; top: 0px; right: 0px; transform: translate(50%, -50%); transform-origin: 100% 0%; z-index: 1;',
    )
    await wrapper.setData({ dot:true })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
