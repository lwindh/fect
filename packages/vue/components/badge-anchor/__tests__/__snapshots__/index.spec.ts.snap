// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeAnchor should be render as a context  1`] = `
<div class="container">
  <div class="fect-badge__anchor">
    <p>Test Message</p><sup style="position: absolute; top: 0px; right: 0px; transform: translate(50%, -50%); transform-origin: 100% 0%; z-index: 1;">
      <div class="fect-dot fect-badge__dot"><span class="fect-dot__circle fect-dot__circle--default"></span><span class="fect-dot__ctx"><!----></span></div>
    </sup>
  </div>
</div>
`;
